### ISSUE TITLE
🐛 Bug: Environment does not exist; Exception while fetching models on Google colab

### ISSUE THREAD

**Issue by UChisom** [ersilia-os/ersilia#372]
### Describe the bug.

I tried to fetch the models `eos4tcc`, `eos2ta5`, and `eos6m4j `using the google colab format. For all of the models I got thesame exception: environment does not exist. 

However, on the first run of `eos4tcc` I got an `emptyoutput exception` after a very lengthy error message body. 

Only one of my models fetched successfully

### Describe the steps to reproduce the behavior

`!ersilia -v fetch eos4tcc`

### Expected behavior.

Model fetch successful

### Screenshots.

[eos4tcc 1st run colab.pdf](https://github.com/ersilia-os/ersilia/files/9804721/eos4tcc.1st.run.colab.pdf)
[eos4tcc 2nd run colab.log](https://github.com/ersilia-os/ersilia/files/9804727/eos4tcc.2nd.run.colab.log)
[eos6m4j colab.log](https://github.com/ersilia-os/ersilia/files/9804732/eos6m4j.colab.log)
[eos2t5 colab.log](https://github.com/ersilia-os/ersilia/files/9804735/eos2t5.colab.log)

### Operating environment

Ubuntu 22.04.1 LTS

### Additional context

Only model `eos3d3r `ran successfully and made a prediction. 
This was also the only model to successfully run in my Ubuntu terminal

**Comment by UChisom** [comment]
@GemmaTuron kindly assist.

**Comment by GemmaTuron** [comment]
Hi @UChisom !

We have been discussing this with @pauline-banye and @malikbadmus on  #366 and seems we have been able to get a temporary fix while I dig more into the root cause of this. See also #378 !
Let us know if this solves the issues for you as well

**Comment by UChisom** [comment]
Hi @GemmaTuron thanks. I'm working through possible solutions now. I'll let you know how it goes.

**Comment by UChisom** [comment]
. @GemmaTuron I have tried the solution by Malikbadmus, but I still get the errors as described in the log files. 
[eos6m4j_.log](https://github.com/ersilia-os/ersilia/files/9814280/eos6m4j_.log)
[eos4tcc_.log](https://github.com/ersilia-os/ersilia/files/9814282/eos4tcc_.log)

I really need help here. Three of my models haven't been able to run either on Ubuntu or Colab. @zainab-ik also experienced similar issues with the same models

**Comment by Zainab-ik** [comment]
@GemmaTuron While replicating Malik solution, the model still outputs same error. Also after deleting runtime and reconnecting.

**Comment by pauline-banye** [comment]
Hi @GemmaTuron I'm done with the colab [template](https://colab.research.google.com/drive/1XvEil6gJGuiHdqID9jpubE9gVHcdxRUF#scrollTo=1cg9EqzVnJSs).

 Please review.

**Comment by GemmaTuron** [comment]
> . @GemmaTuron I have tried the solution by Malikbadmus, but I still get the errors as described in the log files. [eos6m4j_.log](https://github.com/ersilia-os/ersilia/files/9814280/eos6m4j_.log) [eos4tcc_.log](https://github.com/ersilia-os/ersilia/files/9814282/eos4tcc_.log)
> 
> I really need help here. Three of my models haven't been able to run either on Ubuntu or Colab. @Zainab-ik also experienced similar issues with the same models

Hi  @UChisom just to confirm, these are logs form Google Colab?

**Comment by GemmaTuron** [comment]
> 

Hi @Zainab-ik ,
I understand your frustration, let's park eos6m4j, but I'd like to fix the problems on the others, for what i can see in the excel:

- eos2ta5: gives problems to both @UChisom and @Zainab-ik (but runs ok in my computer)
- eos4tcc: same as above
- eos3d3r: @UChisom succedded, @Zainab-ik  could not fetch
Seeing this, maybe lets start by eos3d3r. Is there a specific log file for that one?
And can you both point me to the issues where you have pasted info about the eos2ta5 and eos4tcc? 
Thanks for your patience, and dont worry about getting stuck, we can progress while keeping these on the backlog and trying to debug them ;)

**Comment by UChisom** [comment]
Yes, @GemmaTuron they are all logs from colab.

Issue #356 contains info on `eos2ta5`, while #357 contains that on `eos4tcc`. Both when run on Ubuntu

Thanks for the encouragement 😅

**Comment by UChisom** [comment]
To be clear, we're leaving out eos6m4j right?

**Comment by Zainab-ik** [comment]
> To be clear, we're leaving out eos6m4j right?

for now

**Comment by Zainab-ik** [comment]
> > 
> 
> Hi @Zainab-ik , I understand your frustration, let's park eos6m4j, but I'd like to fix the problems on the others, for what i can see in the excel:
> 
> * eos2ta5: gives problems to both @UChisom and @Zainab-ik (but runs ok in my computer)
> * eos4tcc: same as above
> * eos3d3r: @UChisom succedded, @Zainab-ik  could not fetch
>   Seeing this, maybe lets start by eos3d3r. Is there a specific log file for that one?
>   And can you both point me to the issues where you have pasted info about the eos2ta5 and eos4tcc?
>   Thanks for your patience, and dont worry about getting stuck, we can progress while keeping these on the backlog and trying to debug them ;)

@GemmaTuron Thanks for the clarification.
#349 reference the `eos2ta5` while #389 reference `eos4tcc` . Also, I've been able to fetch `eos3d3r` on colab but not on CLI 
[eos3d3rE.log](https://github.com/ersilia-os/ersilia/files/9827176/eos3d3rE.log), this is the log file from CLI

Kindly look through

**Comment by GemmaTuron** [comment]
Hi @Zainab-ik !

Thanks. Definitely there is an issue with your connection, see this thread: https://stackoverflow.com/questions/45576930/python-requests-newconnectionerror
Can you try to ping google from your CLI and see if it responds? This[ thread](https://stackoverflow.com/questions/52815784/python-pip-raising-newconnectionerror-while-installing-libraries) is suggesting a solution that also works for WSL and virtual environments, so definitely let's try this out!
There are two things in this issue:
- DNS not properly configured
- Docker messing things up. Did you install docker? see in the thread there is a comment about needing to restart docker.
I suggest:

1. Make sure your PROXY is off -- can you confirm that?
2. 1. If you have docker installed, restart it as explained in the stackoverflow: ```
systemctl restart NetworkManager.service
sudo service docker restart
```
3. Try adding the DNS server for google


All these are suggested on the stackoverflow, try one by one and see which one hopefully helps!

**Comment by GemmaTuron** [comment]
@UChisom in your case, the log on eos4tcc is pointing to this annoying tqdm issue, which is already been reported elsewhere #389 , so let's leave that for the moment.
And eos6mj is in #387 , so all errors reported !
Let's use this issue thread to try and solve the eos3d3r issue with @Zainab-ik, it would be great to identify how to solve the NewConnectionError so we can help future users

**Comment by UChisom** [comment]
Thanks for the feedback @GemmaTuron 

I've read through the references.
I'll mark the other models in purple on the sheet excluding eos3d3r.

I'm reviewing the error in @Zainab-ik log file. I've had similar issues with other models, but I haven't cracked yet. I'll read more and revert!

**Comment by GemmaTuron** [comment]
Hi, I am also adding here @EstherIdabor because she also gets the network connection error. It would be great if you can do some research with @Zainab-ik, starting for this  https://stackoverflow.com/questions/52815784/python-pip-raising-newconnectionerror-while-installing-libraries

**Comment by UChisom** [comment]
Noted @GemmaTuron .

**Comment by Zainab-ik** [comment]
Thanks @GemmaTuron

**Comment by Zainab-ik** [comment]
> Hi @Zainab-ik !
> 
> Thanks. Definitely there is an issue with your connection, see this thread: https://stackoverflow.com/questions/45576930/python-requests-newconnectionerror Can you try to ping google from your CLI and see if it responds? This[ thread](https://stackoverflow.com/questions/52815784/python-pip-raising-newconnectionerror-while-installing-libraries) is suggesting a solution that also works for WSL and virtual environments, so definitely let's try this out! There are two things in this issue:
> 
> * DNS not properly configured
> * Docker messing things up. Did you install docker? see in the thread there is a comment about needing to restart docker.
>   I suggest:
> 
> 1. Make sure your PROXY is off -- can you confirm that?
> 2. 1. If you have docker installed, restart it as explained in the stackoverflow:
> 
> ```
> systemctl restart NetworkManager.service
> sudo service docker restart
> ```
> 
> 3. Try adding the DNS server for google
> 
> All these are suggested on the stackoverflow, try one by one and see which one hopefully helps!

Thanks @GemmaTuron 

- ping google.com on my CLI works normally, also I tried ping 8.8.8.8 also as suggested in the link, it also worked. 
- I changed my DNS configuration to that google for both IPv4 and IPv6 also and rerun the model, same error persists.
- I uninstalled docker and installed again also.


I'd try the proxy and give feedback but above methods didn't seem to solve the issue for me.

**Comment by UChisom** [comment]
Hi @GemmaTuron, I'd like to ask if not having docker set up on your PC be a probable cause of any error.

**Comment by GemmaTuron** [comment]
Hi @UChisom,

Docker is as of now not a hard requirement for running ersilia (will be in the future). So, not having docker installed should not be a problem. Having docker installed but not properly might be a problem though!
@Zainab-ik try the proxy and let me know. IF none of this works, we will find a time where we can both connect and try to solve this together, we are really interested in understanding what configuration is causing this.

**Comment by UChisom** [comment]
Oh okay, thanks @GemmaTuron 

I have a suggestion for @Zainab-ik on `eos3d3r`
Could she suppress the __Unverified HTTPS request is being made to host__ warnings using the warning flag `-W` and see if it runs?

**Comment by Zainab-ik** [comment]
> Oh okay, thanks @GemmaTuron
> 
> I have a suggestion for @Zainab-ik on `eos3d3r` Could she suppress the **Unverified HTTPS request is being made to host** warnings using the warning flag `-W` and see if it runs?

how do I do that?

**Comment by Zainab-ik** [comment]
> Hi @Zainab-ik !
> 
> Thanks. Definitely there is an issue with your connection, see this thread: https://stackoverflow.com/questions/45576930/python-requests-newconnectionerror Can you try to ping google from your CLI and see if it responds? This[ thread](https://stackoverflow.com/questions/52815784/python-pip-raising-newconnectionerror-while-installing-libraries) is suggesting a solution that also works for WSL and virtual environments, so definitely let's try this out! There are two things in this issue:
> 
> * DNS not properly configured
> * Docker messing things up. Did you install docker? see in the thread there is a comment about needing to restart docker.
>   I suggest:
> 
> 1. Make sure your PROXY is off -- can you confirm that?
> 2. 1. If you have docker installed, restart it as explained in the stackoverflow:
> 
> ```
> systemctl restart NetworkManager.service
> sudo service docker restart
> ```
> 
> 3. Try adding the DNS server for google
> 
> All these are suggested on the stackoverflow, try one by one and see which one hopefully helps!



> Hi @UChisom,
> 
> Docker is as of now not a hard requirement for running ersilia (will be in the future). So, not having docker installed should not be a problem. Having docker installed but not properly might be a problem though! @Zainab-ik try the proxy and let me know. IF none of this works, we will find a time where we can both connect and try to solve this together, we are really interested in understanding what configuration is causing this.

Hi @GemmaTuron , I disabled my proxy also, reinstall all packages and tried fetching again, I got a `TypeError: expected str, bytes or os.PathLike object, not NoneType`. Just the same error in the error log.

**Comment by Zainab-ik** [comment]
i dont know what else to try again

**Comment by UChisom** [comment]
> > Oh okay, thanks @GemmaTuron
> > I have a suggestion for @Zainab-ik on `eos3d3r` Could she suppress the **Unverified HTTPS request is being made to host** warnings using the warning flag `-W` and see if it runs?
> 
> how do I do that?

@Zainab-ik 
I haven't been able to use the flag yet, but it may be used as one of the following: see [link](https://docs.python.org/3/using/cmdline.html#cmdoption-W)
And I'm suggesting using the `-Wignore` @GemmaTuron please can you assist.

**Comment by UChisom** [comment]
> i dont know what else to try again

Dees the new installation now work?

**Comment by GemmaTuron** [comment]
Hi @Zainab-ik !

Let me know if this last fix by @UChisom is working, otherwise we will schedule a quick call to try and solve it together.

**Comment by Zainab-ik** [comment]
Update (Goodnews)

Hi, @GemmaTuron and @UChisom 
I've been able to fetch and predict the model `eos3d3r` on my CLI. While trying different configurations, my CLI came up with a `systemd` error and stopped working. I had to repair and reset and unfortunately uninstall.

Steps to fetch successfully, 

- I did a new installation of `ubuntu on windows` and updated it using the `WSL` to `ubuntu 20.04.4`.
-  I started the Ersilia installation all over. 

I'd like to suggest an edit to the installation guide:

- The installation of GitHub CLI and GitHub LFS should be done after installing Ersilia.

I realized installing both packages in the terminal outside the Ersilia environment makes them unrecognized. 

- I also noticed, my former ubuntu had 86% used-up memory while the new installation had just 0.1% used-up memory. 
- I suggest memory should be considered when installing and in cases of memory, it should be freed for a smooth process.

- While rerunning the models also, I disabled the PROXY.

finally, I've been able to predict. 
Thank you @GemmaTuron @UChisom
